src/containers/notification/store/sagas.ts

```diff
@@ -552,6 +551,13 @@ function* resetNotificationBadgeCount() {
   }
 }
 
+// On Native App open and enter foreground, clear the notification badges
+function* watchResetNotificationBadgeCount() {
+  yield call(waitForBackendSetup)
+  yield call(resetNotificationBadgeCount)
+  yield takeEvery(MessageType.ENTER_FOREGROUND, resetNotificationBadgeCount)
+}
+
 export default function sagas() {
   const sagas: (() => Generator)[] = [
```

Comment:
im surprised you needed to type this

Reply:
Yeah the generator takes various args using the Iterator interface and it was complaining because different generators had differing signatures.